lib/spack/spack/build_systems/rocm.py

```diff
@@ -154,39 +98,66 @@ def hip_flags(amdgpu_target):
         archs = ",".join(amdgpu_target)
         return "--amdgpu-target={0}".format(archs)
 
-    # HIP version vs Architecture
-
-    # TODO: add a bunch of lines like:
-    # depends_on('hip@:6.0', when='amdgpu_target=gfx701')
-    # to indicate minimum version for each architecture.
-
-    # Add compiler minimum versions based on the first release where the
-    # processor is included in llvm/lib/Support/TargetParser.cpp
-    depends_on("llvm-amdgpu@4.1.0:", when="amdgpu_target=gfx900:xnack-")
-    depends_on("llvm-amdgpu@4.1.0:", when="amdgpu_target=gfx906:xnack-")
-    depends_on("llvm-amdgpu@4.1.0:", when="amdgpu_target=gfx908:xnack-")
-    depends_on("llvm-amdgpu@4.1.0:", when="amdgpu_target=gfx90c")
-    depends_on("llvm-amdgpu@4.3.0:", when="amdgpu_target=gfx90a")
-    depends_on("llvm-amdgpu@4.3.0:", when="amdgpu_target=gfx90a:xnack-")
-    depends_on("llvm-amdgpu@4.3.0:", when="amdgpu_target=gfx90a:xnack+")
-    depends_on("llvm-amdgpu@5.2.0:", when="amdgpu_target=gfx940")
-    depends_on("llvm-amdgpu@4.5.0:", when="amdgpu_target=gfx1013")
-    depends_on("llvm-amdgpu@3.8.0:", when="amdgpu_target=gfx1030")
-    depends_on("llvm-amdgpu@3.9.0:", when="amdgpu_target=gfx1031")
-    depends_on("llvm-amdgpu@4.1.0:", when="amdgpu_target=gfx1032")
-    depends_on("llvm-amdgpu@4.1.0:", when="amdgpu_target=gfx1033")
-    depends_on("llvm-amdgpu@4.3.0:", when="amdgpu_target=gfx1034")
-    depends_on("llvm-amdgpu@4.5.0:", when="amdgpu_target=gfx1035")
-    depends_on("llvm-amdgpu@5.2.0:", when="amdgpu_target=gfx1036")
-    depends_on("llvm-amdgpu@5.3.0:", when="amdgpu_target=gfx1100")
-    depends_on("llvm-amdgpu@5.3.0:", when="amdgpu_target=gfx1101")
-    depends_on("llvm-amdgpu@5.3.0:", when="amdgpu_target=gfx1102")
-    depends_on("llvm-amdgpu@5.3.0:", when="amdgpu_target=gfx1103")
-
-    # Compiler conflicts
-
-    # TODO: add conflicts statements along the lines of
-    # arch_platform = ' target=x86_64: platform=linux'
-    # conflicts('%gcc@5:', when='+cuda ^cuda@:7.5' + arch_platform)
-    # conflicts('platform=darwin', when='+cuda ^cuda@11.0.2:')
-    # for hip-related limitations.
+    @classmethod
+    def rocm_version(cls, spec):
+        version = None
+        if "rocm-config" in spec:
+            version = spec["rocm-config"].version
+        return version
+
+    @classmethod
+    def get_hip_flags(cls, spec):
+        return ROCmPackage.hip_flags(cls.get_amdgpu_targets(spec))
+
+    # Get the config amdgpu_targets
+    @classmethod
+    def get_amdgpu_targets(cls, spec):
+        targets = {}
+        if "rocm-config" in spec:
+            targets = spec["rocm-config"].variants["amdgpu_target"].value
+            for _target in spec["rocm-config"].variants["amdgpu_target"].value:
+                targets.add(_target.split(":")[0])
+        return [x for x in targets]
+
+
+    # Get the config amdgpu_targets without feature strings
+    @classmethod
+    def get_amdgpu_targets_with_features(cls, spec, features="all"):
```

Comment:
There is a mismatch between the comment (`without`) and the function name (`with`).